src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +263,216 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
             return enabled;
         }
 
+        public ValueTask<Variant> GetVariantAsync(string feature, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(feature))
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            return GetVariantAsync(feature, null, false, cancellationToken);
+        }
+
+        public ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(feature))
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            return GetVariantAsync(feature, context, true, cancellationToken);
+        }
+
+        private async ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, bool useContext, CancellationToken cancellationToken)
+        {
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider
+                .GetFeatureDefinitionAsync(feature)
+                .ConfigureAwait(false);
+
+            if (featureDefinition == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingFeature,
+                    $"The feature declaration for the feature '{feature}' was not found.");
+            }
+
+            if (featureDefinition.Allocation == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingAllocation,
+                    $"No allocation is defined for the feature {featureDefinition.Name}");
+            }
+
+            if (!featureDefinition.Variants?.Any() ?? false)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingVariantDefinition,
+                    $"No variants are registered for the feature {feature}");
+            }
+
+            VariantDefinition variantDefinition;
+
+            bool isFeatureEnabled = await IsEnabledAsync(feature, context, true, cancellationToken).ConfigureAwait(false);
+
+            variantDefinition = await GetVariantDefinitionAsync(featureDefinition, context, useContext, isFeatureEnabled, cancellationToken).ConfigureAwait(false);
+
+            if (variantDefinition == null)
+            {
+                return null;
+            }
+
+            IConfigurationSection variantConfiguration = null;
+
+            bool configValueSet = !string.IsNullOrEmpty(variantDefinition.ConfigurationValue);
+            bool configReferenceValueSet = !string.IsNullOrEmpty(variantDefinition.ConfigurationReference);
+
+            if (configValueSet && configReferenceValueSet)
```

Comment:
We're straying a bit away from throwing. How about we let inline configuration when?

Reply:
we'll want to add a test to validate the behavior though.

Reply:
Ok I can change the exceptions test I had to validate these scenarios we're changing.